index.js

```diff
@@ -81,6 +88,26 @@ module.exports = function (kibana) {
             }
           }
 
+          if (config.has('oauth2.allowedOrganizations')) {
+            const allowedOrganizations = config.get('oauth2.allowedOrganizations');
+            const token = _.get(request.auth.credentials, 'token');
+            axios.get('https://api.github.com/user/orgs', {
+              headers: { 'Authorization': 'token ' + token }
+            }).then(function(response) {
+              const organizations = response.data.map((org) => org.login);
+              for (let organization of organizations) {
+                if (allowedOrganizations.indexOf(organization) !== -1) {
+                  request.auth.session.set(request.auth.credentials);
+                  return reply.redirect('./');
+                }
+              }
+              return reply(Boom.forbidden('None of the user\'s organization is allowed'));
+            }).catch(function (error) {
+              return reply(Boom.unauthorized('Unable to verify the user\'s organizations'));
+            });
+            return;
```

Comment:
I'm not familiar with hapi or the kibana plugin ecosystem.
It should be good now.